src/llama_recipes/utils/train_utils.py

```diff
@@ -24,14 +25,48 @@
 from llama_recipes.policies import fpSixteen,bfSixteen, get_llama_wrapper
 from llama_recipes.utils.memory_utils import MemoryTrace
 from accelerate.utils import is_xpu_available, is_ccl_available
-
+from llama_recipes.utils.flop_utils import FlopMeasure
 def set_tokenizer_params(tokenizer: LlamaTokenizer):
     tokenizer.pad_token_id = 0
     tokenizer.padding_side = "left"
 
-# Converting Bytes to Megabytes
-def byte2mb(x):
-    return int(x / 2**20)
+@contextlib.contextmanager
+def throughput_measure_context(cfg, local_rank=None):
+    use_profiler: bool = cfg.use_profiler
+    use_flop_counter: bool = cfg.flop_counter
+    if use_flop_counter and use_profiler:
+        raise ValueError("Cannot use both profiler and flop counter")
+    if use_profiler:
```

Comment:
design wise, it seems we need to have different context managers for Flop count and profiling, shall we separate the two, and change the class name from throughput to Flop count or something along that line.

Reply:
Sure, changing "def throughput_measure_context(cfg, local_rank=None)" to "def profile_context(cfg, local_rank=None)".